### PR TITLE
[Remove Stream usage: portal-tools-service-builder ]Lps 176213

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-176213

### CODE REVIEW

File: modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java [hunk 1]
```diff
@@ -5895,14 +5895,24 @@ private boolean _hasFinderThatIsNotUniqueByCompany(
...
+			finderColumnElements,
+			finderColumnElement -> {
+				String finderColumnName = finderColumnElement.attributeValue(
+					"name");
+
+				if (finderColumnName == null) {
+					return null;
+				}
+
+				if (finderColumnName.endsWith("Id") ||
```
Comment (dantewang) [inline modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5908]: @slnn Please flip this `if`, we usually put `return null` in shortcut.

**Comment by slnn** [comment]
ci:test:sf

**Comment by slnn** [comment]
ci:test:relevant

File: modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java (hunk 1 above)
Comment (slnn) [reply modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5911]: Done

**Review by dantewang — APPROVED** [review]

**Comment by Kvale1733** [comment]
@slnn Compile issue in service-builder module
```
 [exec] > Task :util:portal-tools-service-builder:compileJava FAILED
     [exec] 
     [exec] > Task :util:portal-tools-service-builder:copyLibs SKIPPED
     [exec] > Task :util:portal-tools-service-builder:copyCSS NO-SOURCE
     [exec] > Task :util:portal-tools-service-builder:buildCSS NO-SOURCE
     [exec] /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5900: error: lambda body is neither value nor void compatible
     [exec] 			finderColumnElement -> {
     [exec] 			^
     [exec] /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5898: error: no suitable method found for transform(java.util.List<org.dom4j.Element>,(finderCol[...]; } })
     [exec] 		List<String> finderColumnNames = TransformUtil.transform(
     [exec] 		                                              ^
     [exec]     method com.liferay.petra.function.transform.TransformUtil.<T,R,E>transform(java.util.Collection<T>,com.liferay.petra.function.UnsafeFunction<T,R,E>) is not applicable
     [exec]       (cannot infer type-variable(s) T,R,E
     [exec]         (argument mismatch; bad return type in lambda expression
     [exec]           missing return value))
     [exec]     method com.liferay.petra.function.transform.TransformUtil.<T,R,E>transform(T[],com.liferay.petra.function.UnsafeFunction<T,R,E>,java.lang.Class<?>) is not applicable
     [exec]       (cannot infer type-variable(s) T,R,E
     [exec]         (actual and formal argument lists differ in length))
     [exec] Note: /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java uses or overrides a deprecated API.
     [exec] Note: Recompile with -Xlint:deprecation for details.
     [exec] Note: /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/maven/BuildServiceMojo.java uses unchecked or unsafe operations.
     [exec] Note: Recompile with -Xlint:unchecked for details.
```
[Log](https://testray.liferay.com/reports/production/logs/2023-02/test-1-38/test-portal-acceptance-pullrequest(master)/6554/modules-compile-jdk8/0/13/jenkins-console.txt.gz?authuser=0)

**Comment by slnn** [comment]
> @slnn Compile issue in service-builder module
> 
> ```
>  [exec] > Task :util:portal-tools-service-builder:compileJava FAILED
>      [exec] 
>      [exec] > Task :util:portal-tools-service-builder:copyLibs SKIPPED
>      [exec] > Task :util:portal-tools-service-builder:copyCSS NO-SOURCE
>      [exec] > Task :util:portal-tools-service-builder:buildCSS NO-SOURCE
>      [exec] /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5900: error: lambda body is neither value nor void compatible
>      [exec] 			finderColumnElement -> {
>      [exec] 			^
>      [exec] /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java:5898: error: no suitable method found for transform(java.util.List<org.dom4j.Element>,(finderCol[...]; } })
>      [exec] 		List<String> finderColumnNames = TransformUtil.transform(
>      [exec] 		                                              ^
>      [exec]     method com.liferay.petra.function.transform.TransformUtil.<T,R,E>transform(java.util.Collection<T>,com.liferay.petra.function.UnsafeFunction<T,R,E>) is not applicable
>      [exec]       (cannot infer type-variable(s) T,R,E
>      [exec]         (argument mismatch; bad return type in lambda expression
>      [exec]           missing return value))
>      [exec]     method com.liferay.petra.function.transform.TransformUtil.<T,R,E>transform(T[],com.liferay.petra.function.UnsafeFunction<T,R,E>,java.lang.Class<?>) is not applicable
>      [exec]       (cannot infer type-variable(s) T,R,E
>      [exec]         (actual and formal argument lists differ in length))
>      [exec] Note: /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java uses or overrides a deprecated API.
>      [exec] Note: Recompile with -Xlint:deprecation for details.
>      [exec] Note: /opt/dev/projects/github/liferay-portal/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/maven/BuildServiceMojo.java uses unchecked or unsafe operations.
>      [exec] Note: Recompile with -Xlint:unchecked for details.
> ```
> 
> [Log](https://testray.liferay.com/reports/production/logs/2023-02/test-1-38/test-portal-acceptance-pullrequest(master)/6554/modules-compile-jdk8/0/13/jenkins-console.txt.gz?authuser=0)

Thank @Kvale1733 for catching this, will fix and resend

**Comment by slnn** [comment]
Fix and resend in https://github.com/liferay-core-infra/liferay-portal/pull/3659

**Pull request closed without merging**